core/include/scipp/core/variable.tcc

```diff
@@ -574,6 +577,16 @@ public:
     return std::make_unique<ViewModel<T>>(this->dims(), m_values, m_variances);
   }
 
+  bool isSame(const VariableConcept &other) const override {
+    // Views can be copied but can still refer to same data, so unlike for
+    // DataModel a pointer comparison is not sufficient here.
+    if (hasVariances() != other.hasVariances())
+      return false;
+    if (const auto *ptr = dynamic_cast<const ViewModel<T> *>(&other); ptr)
```

Comment:
Trailing `;  ptr`? Not needed

Reply:
Right, I thought I need the c++17 `if`-init here.